drivers/chatbot/driver.js

```diff
@@ -0,0 +1,243 @@
+'use strict';
+
+const { Driver, Device } = require('homey');
+const { OpenAIApi } = require('openai');
+const { v4: uuidv4 } = require('uuid');
+
+/**
+ * This is the driver for chat bot devices that communicate with
+ * OpenAI APIs to add LLM driven conversation logic for Homey.
+ * 
+ * TODO:
```

Comment:
Added some todos here for future work that will be nice additional features.
I haven't added the text splitting feature from the original api as I wasn't sure I wanted to add that complexity yet.